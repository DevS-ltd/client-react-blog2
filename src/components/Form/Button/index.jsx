import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';

export const Button = ({ type, color, disabled, children, onClick }) => {
	return (
		<button
			type={type}
			className={`btn btn__${color}`}
			disabled={disabled}
			onClick={onClick}
		>
			{children}
		</button>
	);
};

Button.propTypes = {
	type: PropTypes.oneOf(['button', 'submit', 'reset']),
	color: PropTypes.oneOf([
		'primary',
		'secondary',
		'success',
		'danger',
		'warning',
		'link',
	]),
	disabled: PropTypes.bool,
	children: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
		.isRequired,
	onClick: PropTypes.func,
};

Button.defaultProps = {
	type: 'submit',
	color: 'primary',
	disabled: false,
};
