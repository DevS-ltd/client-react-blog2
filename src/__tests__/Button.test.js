import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from '../components/Form/index';

describe('Button component', () => {
	test('Default Button', () => {
		const button = renderer.create(<Button>Primary</Button>);
		expect(button.toJSON()).toMatchSnapshot();
	});

	test('Secondary Button', () => {
		const button = renderer.create(
			<Button color="secondary">Secondary</Button>,
		);
		expect(button.toJSON()).toMatchSnapshot();
	});

	test('Disabled button type Button', () => {
		const button = renderer.create(
			<Button disabled={true} type="button">
				Disabled
			</Button>,
		);
		expect(button.toJSON()).toMatchSnapshot();
	});
});
