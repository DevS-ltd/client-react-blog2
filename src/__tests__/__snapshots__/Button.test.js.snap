// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Default Button 1`] = `
<button
  className="btn btn__primary"
  disabled={false}
  type="submit"
>
  Primary
</button>
`;

exports[`Button component Disabled button type Button 1`] = `
<button
  className="btn btn__primary"
  disabled={true}
  type="button"
>
  Disabled
</button>
`;

exports[`Button component Secondary Button 1`] = `
<button
  className="btn btn__secondary"
  disabled={false}
  type="submit"
>
  Secondary
</button>
`;
